### ISSUE TITLE
called `Result::unwrap()` on an `Err` value: Error { repr: Custom(Custom { kind: Other, error: StringError("MPSC send failed") }) }

### ISSUE THREAD

**Issue by st3fan** [servo/servo#12074]
called `Result::unwrap()` on an `Err` value: Error { repr: Custom(Custom { kind: Other, error: StringError("MPSC send failed") }) }
### URL:
https://blog.jetbrains.com/kotlin/2016/06/kotlin-1-0-3-is-here/
### Backtrace:
```
frame #0  - 0x000000010b1f26ce - backtrace::backtrace::trace::hccde8df28b4db2a2
frame #1  - 0x000000010b1f2651 - backtrace::capture::Backtrace::new::h42f95930bb8c5ee8
frame #2  - 0x0000000109a5fca4 - _<F as alloc..boxed..FnBox<A>>::call_box::haeff945be1a63d60
frame #3  - 0x000000010b166947 - util::panicking::initiate_panic_hook::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h2c0e77c593c00c08
frame #4  - 0x000000010b3da844 - std::panicking::rust_panic_with_hook::h983af77c1a2e581b
frame #5  - 0x000000010b3f3786 - std::panicking::begin_panic::he426e15a3766089a
frame #6  - 0x000000010b3dbe88 - std::panicking::begin_panic_fmt::hdddb415186c241e7
frame #7  - 0x000000010b3f33df - rust_begin_unwind
frame #8  - 0x000000010b41a260 - core::panicking::panic_fmt::hf4e16cb7f0d41a25
frame #9  - 0x0000000109a5138e - core::result::unwrap_failed::hf40cb93f6f31ec3d
frame #10 - 0x0000000109acee0f - layout_thread::LayoutThread::exit_now::h9c02213ba81aad74
frame #11 - 0x0000000109ab23a3 - layout_thread::LayoutThread::handle_request_helper::h1029bbb7e9e50e59
frame #12 - 0x0000000109a9df1a - layout_thread::LayoutThread::handle_request::ha13841b9d3ecdc81
frame #13 - 0x0000000109a5db1a - std::panicking::try::call::h401e4a3a7768c3f1
frame #14 - 0x000000010b3f646b - __rust_try
frame #15 - 0x000000010b3f6405 - __rust_maybe_catch_panic
frame #16 - 0x0000000109a5ee84 - _<F as alloc..boxed..FnBox<A>>::call_box::h8169792cb3dac906
frame #17 - 0x000000010b3f1ee8 - std::sys::thread::Thread::new::thread_start::h9c883b6d445ece46
frame #18 - 0x00007fff82eb199c - _pthread_body
frame #19 - 0x00007fff82eb1919 - _pthread_start

```

**Comment by Ms2ger** [comment]
Should be fixed by #12030.

**Comment by Ms2ger** [comment]
But thanks for your report!
